client/landing/stepper/declarative-flow/onboarding-pm.ts

```diff
@@ -86,16 +87,16 @@ const onboarding: Flow = {
 					navigate( 'siteCreationStep' );
 					return;
 				case 'siteCreationStep':
-					// clearSignupDestinationCookie(); // not sure if this is needed and if it is, where it should go
+					clearSignupDestinationCookie();
 					navigate( 'processing' );
 					return;
 				case 'processing': {
-					// clearSignupDestinationCookie();
+					clearSignupDestinationCookie();
 					const destination = addQueryArgs( '/setup/site-setup/goals', {
 						siteSlug: providedDependencies.siteSlug,
 					} );
 
-					persistSignupDestination( destination ); // not sure if this is needed
+					persistSignupDestination( destination );
```

Comment:
This is used to prevent site duplication. Introduced at #45040 and ported overt to the stepper framework in #68722 as shown below.

https://github.com/Automattic/wp-calypso/blob/3984dc6aff5adf3fc5e77a95b22682e73a5b96da/client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/index.tsx#L34-L52

Reply:
This also acts as a storage for the final signup destination, specially when we redirect to external sites for payment.